### ISSUE TITLE
soil data check, question on soil level  not enough belowground

### ISSUE THREAD

**Issue by fabeit** [EDmodel/ED2#170]
If I set NL%ISOILFLG = 1 then the soil information should be read from a file.
But then why do I get an error message about my soil level depths when the ED2IN file states that "Soil grid and initial conditions if no file is provided", should that part of code in ed_opsec not run if a soil file is provided? Or maybe I am missing something?

**Comment by mpaiao** [comment]
Have you set the variable NL%SOIL_DATABASE correctly? This variable should point to some gridded soil texture dataset, written in hdf5 format, along with a header file containing the grid information. If you want I have a FAO-based map that is global and I can send to you.

**Comment by fabeit** [comment]
Yes Marcos, the variable is set correctly, I am also using the FAO data in fact. In the code I don't see any IF statement that doesn't check the soil levels if  SOIL_DATABASE is provided.

**Comment by mpaiao** [comment]
I'm not sure I understood the problem. The current model reads in the 2-D soil texture map from soil_database, then assigns the same soil texture for all layers, regardless of nzg and slz.

**Comment by fabeit** [comment]
I have to test it but it seems to be that even it has a soil_database input, it still checks that number of soil profiles matches the number of soil profiles described in ED2in even if it will not use them. I had a mismatch between nzg and slz and it stopped the run.

**Comment by mdietze** [comment]
@mpaiao recall that once-upon-a-time ED could use either the FAO (global)or 1-km SSURGO (US only) and the latter did have depth information. The code for loading it was in C and was never ported over from when we purged the C code 8-9 years ago.

**Comment by mpaiao** [comment]
@mdietze Do you think Fabio's error message is some legacy from that time? In any case, the current model can represent different soil textures for each layer.  If anyone is interested in adding back the capability reading in and assigning heterogeneous soil profiles, it should be just a matter of rewriting the initialization routines, everything else is already taken care of.

**Comment by mdietze** [comment]
I'm not 100% sure -- stuck in London Heathrow at the moment and it's not the best place to start digging through code -- but you mentioning that ED shouldn't need to check depth because the soil map doesn't have depth reminded me of that legacy issue, and it seems plausible that this could be part of the bug.

The other place that heterogenous soil layers comes in (besides the ED2IN) is in the .site files, which would allow you to set soil profiles not just for every grid cell individually but also to capture subgrid heterogeneity in the soil profiles. Adding soil profiles to .site is why **I** never bothered to go back and recode the SSURGO module (it was quite a mess if I remember).
